fairlearn/preprocessing/_informationfilter.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.preprocessing import StandardScaler
+
+
+def _scalar_projection(vec, unto):
+    return vec.dot(unto) / unto.dot(unto)
+
+
+def _vector_projection(vec, unto):
+    return _scalar_projection(vec, unto) * unto
+
+
+class InformationFilter(BaseEstimator, TransformerMixin):
+    r"""
```

Comment:
I'm happy with this doc section. @adrinjalali pointed out that it's probably best to always write the documentation along with the code, i.e., in the same PR. That now also applies to the user guide in the docs/user_guide directory. Since we haven't done that so far I'll leave it up to you for now, but if you already have an idea of what that should look like please go ahead and add it.

Some more context: I'm currently making up for what we don't have in terms of existing mitigation technique, metrics, etc., so until that's in a good state I can't really tell anyone else to do things we haven't done.

Reply:
I'd love to write some documents on `fetch_boston` and to show mitigation techniques on that dataset. Before I can do that, the dataset needs to get accepted as well as the technique. I am totally planning on doing this tho. It's my preference to have a seperate PR for it but I suppose I could add it here too. I suspect the PR will take a lot longer though.

Reply:
At least until we're in a better state documentation-wise I'm very much on board with your suggested path forward.